File: projects/blognami/lib/blognami/models/image.rb

require "blognami/models/pageable"

Slick::Database::Row.register :image do

  include :pageable

  def file=(file)
      return if !file.kind_of?(Hash) || !file[:type].to_s.match(/\Aimage\/(.+)\z/)
      self.title = file[:filename]
      self.type = file[:type]
      self.data = file[:tempfile].read
  end

end

File: packages/@blognami/default/lib/models/image.js

export default {
    meta(){
        this.include('pageable');
    },

    set file(file){
        const { filename, mimeType, data } = file;
        const matches = mimeType.match(/^image\/(.+)$/);
        if(matches){
            this.title = filename;
            this.type = matches[1];
            this.data = data;
        }
    }
};
